f5lbaasdriver/test/tempest/tests/api/test_l7policy_rules.py

```diff
@@ -93,7 +97,79 @@ def tearDown(self):
         super(L7PolicyTestJSONBasic, self).tearDown()
 
 
-class L7PolicyJSONReject(L7PolicyTestJSONBasic):
+class L7PolicyTestConstructor(L7PolicyTestJSONBasic):
+    # leave consistency checking to my IDE...
+    rj_1 = 'reject_1'
+    rj_2 = 'reject_2'
+    cnts = 'contains'
+    es = 'es'
+    ireal = 'ireal'
+    real = 'real'
+    starts_with = 'startsWith'
+    ends_with = 'endsWith'
+    wrap_pol = 'wrapper_policy'
+    proj = 'Project_'
+    test = 'test'
+    al = 'al'
+    http_header = "httpHeader"
+    not_ = "not_"
+    qcow2 = 'qcow2'
+    redirect_to_url_1 = 'redirect_to_url_1'
+    ext = 'extension'
+    jpg = 'jpg'
+    _re = 're'
+    redirect_to_pool_1 = 'redirect_to_pool_1'
+    wrapper_policy = ArgSet([wrap_pol], proj)
+    reject_1 = ArgSet([wrap_pol, rj_1], proj)
+    reject_1_contains_al = ArgSet([wrap_pol, rj_1, cnts, al], proj)
+    reject_1_contains_es = ArgSet([wrap_pol, rj_1, cnts, es], proj)
+    reject_1_contains_l = ArgSet([wrap_pol, rj_1, cnts, 'l'], proj)
+    reject_1_endswith_ireal = ArgSet([wrap_pol, rj_1, ends_with, ireal], proj)
+    reject_1_endswith_real = ArgSet([wrap_pol, rj_1, ends_with, real], proj)
+    reject_1_endswith_test = ArgSet([wrap_pol, rj_1, ends_with, test], proj)
+    reject_1_startswith_re = ArgSet([wrap_pol, rj_1, starts_with, _re], proj)
+    reject_1_startswith_real = ArgSet([wrap_pol, rj_1, starts_with, real],
+                                      proj)
+    reject_2 = ArgSet([wrap_pol, rj_2], proj)
+    reject_2_contains_es = ArgSet([wrap_pol, rj_2, cnts, es], proj)
+    reject_2_startswith_real = ArgSet([wrap_pol, rj_2, starts_with, real],
+                                      proj)
+    redirect_to_url_l_httpheader_es = \
+        ArgSet([wrap_pol, redirect_to_url_1, http_header, es], proj)
+    redirect_to_pool_1_extension_qcow2 = \
+        ArgSet([wrap_pol, redirect_to_pool_1, ext, qcow2], proj)
+    redirect_to_pool_1_not__qcow2 = \
+        ArgSet([wrap_pol, redirect_to_pool_1, not_, qcow2], proj)
+    redirect_to_pool_1_extension_jpg = \
+        ArgSet([wrap_pol, redirect_to_pool_1, ext, jpg], proj)
+    redirect_to_pool_1 = ArgSet([wrap_pol, redirect_to_pool_1], proj)
+    redirect_to_url_1 = ArgSet([wrap_pol, redirect_to_url_1], proj)
+
+    @classmethod
+    def assert_with_retry(cls, check, method, arg_set, tenant_id, **kwargs):
+        """Performs an object manipulation"""
+        proj = arg_set.project + tenant_id
+        args = list(arg_set.expected)
+        args.append(proj)
+        try_cnt = 0
```

Comment:
```python
args[0] += "_{}".format(tenant_id)
```
Should perform what you want for adding the tenant_id/listener_id to the wrapper policy.

Reply:
OK I like this retry.